subsys/net/ip/ipv6.c

```diff
@@ -445,6 +445,13 @@ enum net_verdict net_ipv6_process_pkt(struct net_pkt *pkt)
 		log_strdup(net_sprint_ipv6_addr(&hdr->src)),
 		log_strdup(net_sprint_ipv6_addr(&hdr->dst)));
 
+	if (!is_loopback && (net_is_ipv6_addr_loopback(&hdr->dst) ||
+			     net_is_ipv6_addr_loopback(&hdr->src))) {
+		NET_DBG("Dropping ::1 packet");
+		net_stats_update_ipv6_drop(net_pkt_iface(pkt));
```

Comment:
Just noticed this net_stats_udpade_ipv6_drop() could be put at the drop label. Not something for this patch though, it would require its own, since in many places it's calling that then goto drop.

Reply:
Sure, I can send another PR after this one that refactores this part.